### PR TITLE
Cleanup/Refactoring

### PR DESCRIPTION
**Pull request by kalinon**
Thanks for making an awesome lib to use AWS lambdas. While reverse engineering i spent some time refactoring and cleaning things up. Please don't be offended. Really did a good job for a evening hack!

Only improvements i really made was to fix the 1 ameba error and change the HTTP Response json part to hopefully improve it. No more deserializing to serialize it.

### CODE REVIEW

**Comment by kalinon** [comment]
Output of my spec and ameba:
```
➜  crystal-aws-lambda git:(master) ✗ crystal spec
.............

Finished in 1.57 milliseconds
13 examples, 0 failures, 0 errors, 0 pending
➜  crystal-aws-lambda git:(master) ✗ ameba
Inspecting 10 files.

..........

Finished in 22.71 milliseconds

10 inspected, 0 failures.
```

File: shard.yml [hunk 1]
```diff
@@ -10,8 +10,10 @@ development_dependencies:
     branch: master
   ameba:
     github: veelenga/ameba
-    version: 0.8.1
+    version: 0.10.0
+  crelease:
+    github: elorest/CRelease
```
Comment (spinscale) [inline shard.yml]: what exactly do you need this dependency for? I don't see it being used and there is nothing in the `bin/` directory either.

**Comment by spinscale** [comment]
this is an awesome PR. Thanks for putting the effort in to fix my night hacking :-)

I left one last question and then I'll merge it!

File: shard.yml (hunk 1 above)
Comment (kalinon) [reply shard.yml]: Yeah it seems it doesn't build on install like ameba does. crelease is a helper to manage crystal shard versioning and releases. I can remove it.

**Comment by spinscale** [comment]
thanks a lot!

**Pull request merged**